_delphi_utils_python/data_proc/geomap/geo_data_proc.py

```diff
@@ -548,6 +543,99 @@ def derive_chngfips_state_crosswalk():
     group_to_state.to_csv(join(OUTPUT_DIR, CHNGFIPS_STATE_OUT_FILENAME), index=False)
 
 
+def fetch_county_groups_spreadsheet():
+    # County mapping file is derived from
+    # https://docs.google.com/spreadsheets/d/1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0/edit#gid=871427657
+    sheet_id = "1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0"
+    sheet_name = "groupings"
+    # Request sheet in CSV format via tag in URL.
+    url = f"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv&sheet={sheet_name}"
+
+    county_groups = pd.read_csv(
+            url, dtype="string", index_col=False
+        ).dropna(
+            how="all", axis=1
+        )
+    county_groups["state FIPS"] = county_groups["state FIPS"].astype(int)
+
+    # Counties belonging to each group are listed (as FIPS codes) in the "county
+    # FIPS grouping" column, concatenated and separated by the pipe "|". Each
+    # included FIPS code is also listed in its own unnamed column. County groups
+    # can contain varying numbers of counties, and if CHNG provides new or updated
+    # county groupings the number of unnamed columns listing included counties
+    # could change. Use a general approach to find all of them.
+    county_cols = [colname for colname in county_groups.columns if colname.startswith("Unnamed: ")]
+
+    new_names = {
+        "state FIPS": "state_fips",
+        "county FIPS grouping": "fips_list",
+        **{ colname: ("county" + str(i + 1)) for i, colname in enumerate(county_cols)}
```

Comment:
question: are the `countyNN` columns ever used?

iirc those columns were ones I added for some analysis preprocessing, and didn't actually come from CHNG. if something downstream depends on them, we can retain the preprocessing step, but if not we may as well exclude them from the df

Reply:
Oh, I didn't realize the `countyNN` fields weren't in the original spreadsheet. They're used to [turn the spreadsheet into long format](https://github.com/cmu-delphi/covidcast-indicators/blob/ndefries/geomapper/popsafe-county-level/_delphi_utils_python/data_proc/geomap/geo_data_proc.py#L495-L496), but I could also split `fips_list` if we don't want to rely on them being available.

Reply:
I got nerd sniped, so here's split trick, if you didn't get to it yet
```py
sheet_id = "1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0"
sheet_name = "groupings"
# Request sheet in CSV format via tag in URL.
url = f"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv&sheet={sheet_name}"

county_groups = pd.read_csv(
        url, dtype="string", index_col=False
    ).dropna(
        how="all", axis=1
    )
county_groups = county_groups[["popsafe-id", "state FIPS", "county FIPS grouping"]]
county_groups["state FIPS"] = county_groups["state FIPS"].astype(int)
county_groups = pd.concat([county_groups, county_groups["county FIPS grouping"].str.split("|", expand=True)], axis=1)
```